onnxruntime/core/providers/cuda/tensor/tile_impl.cu

```diff
@@ -7,140 +7,178 @@
+#pragma unroll
+  for (int i = 0; i < num_elements_per_thread; ++i) {
+    if (id < N) {
+      output_data[id] = value[i];
+      id += num_threads_per_block;
+    }
   }
-  output_data[id] = input_data[input_index];
 }
 
 template <typename T>
-void TileImpl(
-    cudaStream_t stream,
-    const size_t shape_rank,
-    const TArray<fast_divmod>& fdm_input_shape,
-    const TArray<int64_t>& input_stride,
-    const T* input_data,
-    const TArray<fast_divmod>& fdm_output_strides,
-    T* output_data,
-    const size_t N) {
-  int blocksPerGrid = (int)(ceil(static_cast<float>(N) / GridDim::maxThreadsPerBlock));
-  if (shape_rank > MAX_DIMS) {
-    _TileKernel<T><<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0, stream>>>(
-        shape_rank, fdm_input_shape, input_stride, input_data,
-        fdm_output_strides, output_data, (CUDA_LONG)N);
-  } else {
-    _UnRolledTileKernel<T><<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0, stream>>>(
-        shape_rank, fdm_input_shape, input_stride, input_data,
-        fdm_output_strides, output_data, (CUDA_LONG)N);
-  }
+void TileImpl(cudaStream_t stream, const size_t shape_rank, const TArray<fast_divmod>& fdm_input_shape,
+              const TArray<int64_t>& input_stride, const T* input_data, const TArray<fast_divmod>& fdm_output_strides,
+              T* output_data, const size_t N) {
+  int blocksPerGrid = static_cast<int>(CeilDiv(N, num_threads_per_block * num_elements_per_thread));
+  _UnRolledTileKernel<T><<<blocksPerGrid, num_threads_per_block, 0, stream>>>(shape_rank, fdm_input_shape, input_stride,
+                                                                              input_data, fdm_output_strides,
+                                                                              output_data, static_cast<CUDA_LONG>(N));
 }
 
 template <typename T>
-__global__ void _TileMemcpyKernel(
-    const T* input_data,
-    const size_t num_input_elements,
-    T* output_data,
-    const size_t N) {
+__global__ void _TileMemcpyKernel(const T* input_data, T* output_data, const CUDA_LONG N, const size_t repeats) {
   CALCULATE_ELEMENTWISE_INDEX_OR_EXIT(id, N);
-  auto input_index = id % num_input_elements;
-  output_data[id] = input_data[input_index];
+  T input_val = input_data[id];
+  for (size_t i = 0; i < repeats; ++i) {
+    output_data[id] = input_val;
+    id += N;
+  }
 }
 
 template <typename T>
-void TileMemcpyImpl(
-    cudaStream_t stream,
-    const T* input_data,
-    const size_t num_input_elements,
-    T* output_data,
-    const size_t num_output_elements) {
-  int blocksPerGrid = (int)(ceil(static_cast<float>(num_output_elements) / GridDim::maxThreadsPerBlock));
-  _TileMemcpyKernel<<<blocksPerGrid, GridDim::maxThreadsPerBlock, 0, stream>>>(
-      input_data, num_input_elements, output_data, (CUDA_LONG)num_output_elements);
+void TileMemcpyImpl(cudaStream_t stream, const T* input_data, T* output_data, const size_t num_input_elements,
+                    const size_t repeats) {
```

Comment:
*.cu did not get formatted automatically, shall we make this parameter list follow other kernel formatting. 

"Status SplitImpl(cudaStream_t stream,
                 const size_t element_size,
                 const int block_size_including_axis_dim,
                 const int block_size_inside_axis_dim,
                 const int64_t* split_sizes,
                 const int64_t* split_sizes_range,
                 const int64_t* axis_dimension_input_output_mapping,
                 const int num_outputs,
                 const void* input_data,
                 void** output_ptr,
                 const size_t N) {"

Reply:
cu is just cpp, once you config your VSCode properly (mapping cu file as cpp file), you can format them. I think the best practice is not to worry about the fomatting when coding and let the IDE doing that.

Reply:
well, to me, *.cu is cuda source file, it is reasonable not respecting clang format. it's up to you though.